### ISSUE TITLE
[R-Forge #5670] rolling joins should have a "ignore exact matches" optional argument

### ISSUE THREAD

**Issue by arunsrinivasan** [Rdatatable/data.table#527]
Submitted by: Arun ; Assigned to: Nobody; [R-Forge link](https://r-forge.r-project.org/tracker/index.php?func=detail&aid=5670&group_id=240&atid=978)

When doing a rolling join, if there's an exact match with the key column, then it's returned. But sometimes, it is useful to still get the corresponding date before or after the exact match, even if it exists. 

Here's a nice example where it could be of use: 

http://stackoverflow.com/questions/23414498/create-lagged-vectors-based-on-a-different-data-frame-in-a-panel-in-r

Basically, the values generated as `idx1` and `idx2` in my answer should be possible directly with an argument "exact.match=FALSE" or something of the sort.

There are other posts where I wished we'd this feature as well. I'll try and link them as and when I stumble upon.
